### PR TITLE
 Decorrelate AD from docker to introduce multiple listeners

### PR DESCRIPTION
**Pull request by mfpierre**
### What does this PR do?

* Introduce `ServiceListener` factories, automatically registered via build tags
* Activate AD & config polling by default

### Motivation

This is a necessary step to introduce new AD listeners (Kubelet, ECS, ...)

### CODE REVIEW

File: cmd/agent/common/docker_init.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+package common
+
+import (
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+)
+
+func init() {
```
Comment (mfpierre) [inline cmd/agent/common/docker_init.go]: Opened to suggestions for where to put this

File: cmd/agent/common/autoconfig_nodocker.go [hunk 2]
```diff
@@ -1,56 +0,0 @@
...
-
-	// add the check loaders
-	for _, loader := range loaders.LoaderCatalog() {
-		AC.AddLoader(loader)
-		log.Debugf("Added %s to AutoConfig", loader)
-	}
-
-	// Add the configuration providers
-
-	// BUG(massi): configuration providers should not depend on the `docker` build tag.
```
Comment (mfpierre) [inline cmd/agent/common/autoconfig_nodocker.go]: @masci I'm fixing this I guess

File: cmd/agent/common/docker_init.go (hunk 1 above)
Comment (masci) [reply cmd/agent/common/docker_init.go]: LGTM

File: pkg/collector/listeners/types.go [hunk 3]
```diff
@@ -37,3 +30,30 @@ type ServiceListener interface {
...
+	}
+	_, registered := ServiceListenerFactories[name]
+	if registered {
+		log.Errorf("Service listener factory %s already registered. Ignoring.", name)
+	}
+	ServiceListenerFactories[name] = factory
+}
+
+// DockerService implements and store results from the Service interface for the Docker listener
+type DockerService struct {
```
Comment (masci) [inline pkg/collector/listeners/types.go]: Should this go to its own source module?

**Review by masci — DISMISSED** [review]
Left a comment but really minor stuff, code LGTM 👍

**Review by hkaj — APPROVED** [review]

**Pull request merged**